src/Libraries/CoreNodes/Data.cs

```diff
@@ -103,5 +220,226 @@ public override bool CanConvert(Type objectType)
             }
         }
 
+        private class DesignScriptGeometryConverter : JsonConverter
+        {
+            public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+            {
+                string serializedValue;
+
+                switch(value)
+                {
+                    case Geometry item:
+                        var geoString = item.ToJson();
+
+                        if (!string.IsNullOrEmpty(geoString))
+                        {
+                            writer.WriteRawValue(geoString);
+                            return;
+                        }
+                        break;
+                    case BoundingBox item:
+                        writer.WriteRawValue(item.ToJson());
+                        return;
+                    case CoordinateSystem item:
+                        writer.WriteRawValue(item.ToJson());
+                        return;
+                    case Mesh item:
+                        writer.WriteRawValue(item.ToJson());
+                        return;
+                    case Vector item:
+                        writer.WriteRawValue(item.ToJson());
+                        return;
+                }
+
+                throw new NotSupportedException();
+            }
+
+            public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+            {
+                throw new NotImplementedException("Unnecessary because CanRead is false. The type will skip the converter.");
+            }
+
+            public override bool CanRead
+            {
+                get { return false; }
+            }
+
+            public override bool CanConvert(Type objectType)
+            {
+                return typeof(DesignScriptEntity).IsAssignableFrom(objectType);
+            }
+        }
+
+        private class ColorConveter : JsonConverter
```

Comment:
why use a bunch of converters for this instead of decorating the class with attributes. I guess I get a bit of a code smell when the serialization and deserialization are handled in different places, one using a converter and one using regular constructors. Can it be unified?

Reply:
I think the main reason is the overlap with the node -> file serialization and the node object -> data serialization.  I hope to introduce the interface so that it can be explicit.

Reply:
Nevermind.... file stuff doesn't apply to ZT.  But would be handled by the interface